### PR TITLE
Test implementation

### PR DESCRIPTION
**Pull request by tomat-suppe**
* Implemented 4 tests, whereof 3 test the program and 1 is a baseline test that should never fail and is independent of the program
* The last test is a 'Canary in the Coal Mine' test: we KNOW if this one test, something is wrong with our testing in general
* Modified build-and-test workflow to test the new test directories
* Modified build-and-test workflow to have a more appropriate, descriptive name

Side-quest:
* Modified Cheep and CheepDTO so Text does not only have maxlength rule but also minimumlength (1) rule.

### CODE REVIEW

File: test/Chirp.ChirpInfrastructure.Tests/UnitTests.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+namespace Chirp.ChirpInfraStructure.Tests{
+
+public class UnitTest
+{
+    
+    [Fact]
+    public void TestThatTestWork()
+    {
+       //Arrange
+        string h = "Hej";
```
Comment (Thyme-Cat) [inline test/Chirp.ChirpInfrastructure.Tests/UnitTests.cs:13]: More of a for-funsies comment than a problem comment, but we could rename the string to be "Hello World!" as that's a classic programming sentence. But again it doesn't change the function of the test, so more of a nice to have than need to have.

**Review by Thyme-Cat — APPROVED** [review]
Lgtm, good job

**Review by JaSNOHo — APPROVED** [review]
lgtm

File: src/ChirpCore/Domain/Cheep.cs [hunk 2]
```diff
@@ -11,7 +11,8 @@ public class Cheep
 		public int CheepId { get; set; }
 		public int AuthorId { get; set; }
 		public required Author Author { get; set; }
-		[StringLength(160)] public required string Text { get; set; }
+		[StringLength(160, MinimumLength = 1)]
```
Comment (HenrijetteOest) [inline src/ChirpCore/Domain/Cheep.cs:14]: [Required] [StringLength(160)]

File: src/ChirpCore/DTOs/CheepDTO.cs [hunk 3]
```diff
@@ -9,7 +9,7 @@ namespace ChirpCore.DTOs
 	public class CheepDTO
 	{
 		public int AuthorID { get; set; }
-		[StringLength(160)] public string? Text { get; set; }
+		[StringLength(160, MinimumLength = 1)] public string? Text { get; set; }
```
[the remark this answers is not in the document]
Comment (tomat-suppe) [reply src/ChirpCore/DTOs/CheepDTO.cs:12]: Needs not fixed in DTO

File: src/ChirpCore/Domain/Cheep.cs (hunk 2 above)
Comment (tomat-suppe) [reply src/ChirpCore/Domain/Cheep.cs:14]: Does not need to add required in the DTO :)

**Review by HenrijetteOest — APPROVED** [review]
test folders with or wihtout sln files, at the moment that is the big question, but it does not make any trouble at the moment and we are aware of it now, disgussing it 🥉

**Comment by tomat-suppe** [comment]
> test folders with or wihtout sln files, at the moment that is the big question, but it does not make any trouble at the moment and we are aware of it now, disgussing it 🥉

lets be mindful when working further on tests: maybe shared sln file for tests!

**Review by llsa-itu — APPROVED** [review]
lgtm :)

**Pull request merged**